rskj-core/src/test/java/co/rsk/core/bc/TransactionPoolImplTest.java

```diff
@@ -710,6 +711,52 @@ public void checkTxWhichCanNotBePaidIsRejected() {
         Assert.assertTrue(transactionPool.getQueuedTransactions().isEmpty());
     }
 
+    @Test
+    public void aNewTxIsAddedInTxPoolAndShouldBeAddedInCache(){
+        Coin balance = Coin.valueOf(1000000);
+        createTestAccounts(2, balance);
+        Account account1 = createAccount(1);
+        Transaction tx = createSampleTransaction(1, 2, 1000, 0);
+        transactionPool.addTransaction(tx);
+
+        Assert.assertTrue(signatureCache.containsTx(tx));
+        Assert.assertArrayEquals(signatureCache.getSender(tx).getBytes(), account1.getAddress().getBytes());
+    }
+
```

Comment:
could you add a spy to signatureCache = rskTestContext.getReceivedTxSignatureCache();
And create a test where you add two times the same transaction, then check that the cached called only once the storeSender for that transaction and the getSender was called at least twice

Reply:
i will do this in "integration" tests

Reply:
ok